tests/unit/test_prepend_bos.py

```diff
@@ -0,0 +1,77 @@
+import pytest
+
+from transformer_lens import HookedTransformer
+
+
+class TestPrependBos:
+    prompt = "Hello world!"
+
+    @pytest.fixture(scope="class")
+    def model(self):
+        return HookedTransformer.from_pretrained("gpt2")
+
+    def test_default_prepend_bos_value(self):
+        model = HookedTransformer.from_pretrained("gpt2")
+        assert model.prepend_bos == True, "Default prepend_bos should be True"
+
+    def test_set_prepend_bos(self, model):
+        model.set_prepend_bos(False)
+        assert model.prepend_bos == False, "prepend_bos should be set to False"
+
+    def test_default_prepend_bos_at_method_level(self):
+        model = HookedTransformer.from_pretrained("gpt2")
+
+        logits = model(self.prompt)  # [batch pos d_vocab]
+        str_tokens = model.to_str_tokens(self.prompt)
+        tokens = model.to_tokens(self.prompt)
+
+        assert logits.shape[1] == len(str_tokens) == tokens.shape[1]
```

Comment:
Can you explicitly test that this equals 3? (Assuming that "Hello world" tokenizes as BOS, 'Hello', ' World'). As is, maybe it _never_ prepends BOS

Reply:
I would create a function for "num_tokens_in_prompt" that takes in a prepend_bos Bool and returns the number of tokens in self.prompt, along with the prepend_bos (you can just hard-code these numbers IMO), and then calls in the equals expressions in the tests